src/dwarf_eh.h

```diff
@@ -48,6 +48,18 @@
 #	include "unwind.h"
 #endif
 
+#ifdef __CHERI_PURE_CAPABILITY__
+static inline intptr_t assert_pointer_in_bounds(intptr_t value) {
+  assert(__builtin_cheri_tag_get((void*)value));
+  assert(__builtin_cheri_offset_get((void*)value) <=
+         __builtin_cheri_length_get((void*)value));
```

Comment:
Technically `offset == length` is out of bounds...

Reply:
Oops.